### PR TITLE
Use search controller within navigation

### PR DESCRIPTION
**Pull request by ay8s**
Changes the search bar out for UISearchController which sits in the navigation bar.

### CODE REVIEW

**Review by mcritz — APPROVED** [review]
👍🏻

File: SF iOS/SF iOS/Feed/EventsFeedViewController.m [hunk 1]
```diff
@@ -201,19 +199,21 @@ - (void)viewDidLoad {
...
+    self.searchController.searchResultsUpdater = self;
+    self.searchController.searchBar.placeholder = @"Filter";
+    [self.searchController.searchBar sizeToFit];
+    self.searchController.delegate = self;
+    self.searchController.dimsBackgroundDuringPresentation = NO;
+    self.searchController.searchBar.delegate = self;
+    self.navigationItem.searchController = self.searchController;
+    self.navigationItem.hidesSearchBarWhenScrolling = YES;
+    self.definesPresentationContext = YES;
+    self.searchController.hidesNavigationBarDuringPresentation = NO;
```
Comment (rodericj) [inline SF iOS/SF iOS/Feed/EventsFeedViewController.m:212]: I’ve been putting mega line initializer things like this in their own class. Or at least trying to.

**Comment by rodericj** [comment]
Nice to see so many red lines

File: SF iOS/SF iOS/Feed/EventsFeedViewController.m [hunk 2]
```diff
@@ -32,18 +32,15 @@ @interface EventsFeedViewController () <UITableViewDataSource, UITableViewDelega
...
+
 @end
 NS_ASSUME_NONNULL_END
 
 @implementation EventsFeedViewController
 
-#define kSEARCHBARHEIGHT 60
-#define kSEARCHBARMARGIN 12
-#define kTABLEHEADERHEIGHT (2 * kSEARCHBARHEIGHT)
-
```
Comment (rodericj) [inline SF iOS/SF iOS/Feed/EventsFeedViewController.m:46]: Nice to get rid of this

**Comment by rodericj** [comment]
Just built and ran this. it's very nice.

**Review by rodericj — APPROVED** [review]

File: SF iOS/SF iOS/Feed/EventsFeedViewController.m [hunk 3]
```diff
@@ -201,19 +199,21 @@ - (void)viewDidLoad {
 
     [self registerForPreviewingWithDelegate:self sourceView:self.tableView];
     
-    CGRect searchBarRect = CGRectMake(kSEARCHBARMARGIN,
-                                      0,
-                                      self.view.frame.size.width-(kSEARCHBARMARGIN*2),
-                                      kSEARCHBARHEIGHT);
-    [self setupNotificationsButton];
```
Comment (rodericj) [inline SF iOS/SF iOS/Feed/EventsFeedViewController.m:208]: Oh I see

**Pull request merged**